File: GARI_Umbria_2026-08-03/DotNet/AgronicaCoreStd/AgronicaCoreModelsSTD/metaschema/Disciplinari.cs
using AgronicaCoreModelsSTD.anagrafiche;
using AgronicaCoreModelsSTD.baseClass;
using AgronicaCoreModelsSTD.metaschema.utilizzi;
using System;
using System.Collections.Generic;
using System.Text;

namespace AgronicaCoreModelsSTD.metaschema
{
    public class Disciplinare : BaseCodeDescrStr
    {
        //0=non applicabile/nessun disciplinare/reg bio/nessun vincolo normativo
        //1=pubblico
        //2=privato
        public int disciplinarePubblicoPrivato { get; set; }

        public RegolamentoConcimazione regolamentoConcimazione { get; set; }

        public RaggruppamentiColturaliDPI raggruppamentiColturaliDPI { get; set; }

        public GruppoFinalita gruppoFinalita { get; set; }

        public int flagProtetto { get; set; }

        public int idTr { get; set; }
        public IntervalloTemporale validita { get; set; }

        public Disciplinare(string codice) : base(codice,"")
        {

        }

        public Disciplinare() :base() { }

    }
}

File: GARI_Umbria_2026-08-03/GiasNG/src/app/Model/metaschema/Disciplinari.ts
import { BaseCodeDescrStr } from '../baseClass/baseCodeDescrStr';
import { RegolamentoConcimazione } from './RegolamentoConcimazione';
import { RaggruppamentiColturaliDPI } from './RaggruppamentiColturaliDPI';
import { GruppoFinalita } from './utilizzi/GruppoFinalita';
import {IntervalloTemporale} from "../anagrafiche/IntervalloTemporale";

export class Disciplinare extends BaseCodeDescrStr {

    /*0=non applicabile/nessun disciplinare/reg bio/nessun vincolo normativo
    //1=pubblico
    //2=privato*/
    disciplinarePubblicoPrivato: number;

    regolamentoConcimazione: RegolamentoConcimazione;

    raggruppamentiColturaliDPI: RaggruppamentiColturaliDPI;

    gruppoFinalita: GruppoFinalita;

    flagProtetto: number;

    idTr: number;

    validita: IntervalloTemporale;

    constructor(codice: string) {
        super(codice);
    }
}
